### PR TITLE
Expose additional options to Refresh command

### PR DESCRIPTION
**Pull request by JasonWhall**
[Related Issue](https://github.com/pulumi/pulumi/issues/11336)

Exposes additional options to the Automation API to the [Pulumi refresh](https://www.pulumi.com/docs/reference/cli/pulumi_refresh/) command

### CODE REVIEW

**Comment by RobbieMcKinstry** [comment]
I see the PR is failing CI; it looks like `PULUMI_ACCESS_TOKEN` isn't set. That's by design; we don't want to leak credentials into community contributions.

Not sure what the status of CI in this repo is: whether we rerun CI ourselves or what. Normally we could `bors try` to test the PR.

**Review by Frassle — COMMENTED** [review]
> I see the PR is failing CI; it looks like PULUMI_ACCESS_TOKEN isn't set. That's by design; we don't want to leak credentials into community contributions.

I'm working on fixing that _right now_. I'll ping here once it's fixed and it should just be a case of rebase to get things green.

**Comment by RobbieMcKinstry** [comment]
Hah, perfect, thank you Fraser! :D

File: sdk/Pulumi.Automation/RefreshOptions.cs [hunk 1]
```diff
@@ -13,5 +15,11 @@ public sealed class RefreshOptions : UpdateOptions
         /// Show config secrets when they appear.
         /// </summary>
         public bool? ShowSecrets { get; set; }
+
+        public bool? SkipPendingCreates { get; set; }
+
+        public bool? ClearPendingCreates { get; set; }
+
+        public List<string>? ImportPendingCreates { get; set; }
```
[the remark this answers is not in the document]
Comment (JasonWhall) [reply sdk/Pulumi.Automation/RefreshOptions.cs]: Agreed, I've added a new class that includes the URN and Provider Id fields.

**Comment by JasonWhall** [comment]
> Hey Jason! Thanks for opening this PR. I appreciate you taking the time to contribute back!
> 
> I don't have any objections to this PR. Normally, it's nice to see test cases when we add new features to Automation API; but this change looks reasonably scoped enough to not require tests.
> 
> Admittedly, I'm quite weak in C# compared to the other Pulumi languages. I'll defer to the experts in C# for additional review.
> 
> **One additional note for reviews:** As a reminder, we've recently added an SLO for ensuring Automation API parity across languages. Looking at the issue history, it looks like we'll need to add these flags to the other languages. I'll double-check and open a tracking issue if/when this PR is merged.

Thanks @RobbieMcKinstry! I would like to contribute tests for this change, however I'd probably need a bit of advice on how best to achieve this. As far as I can tell to test this change we would need to create a stack and interrupt this during creation to allow resource to be in a pending operation in the stack. If there are any examples or guidance on how to achieve this, I can update this PR or raise a separate change to introduce some tests for this.

**Comment by Frassle** [comment]
Hey @JasonWhall just an FYI if you rebase this PR on latest changes to main the tests should be able to pass now if we run them.

File: sdk/Pulumi.Automation/WorkspaceStack.cs [hunk 2]
```diff
@@ -489,6 +489,23 @@ void OnPreviewEvent(EngineEvent @event)
...
+                    args.Add("--skip-pending-creates");
+
+                if (options.ClearPendingCreates is true)
+                    args.Add("--clear-pending-creates");
+
+                if (options.ImportPendingCreates?.Any() == true)
+                {
+                    foreach (var item in options.ImportPendingCreates)
+                    {
+                        args.Add("--import-pending-creates");
```
Comment (Frassle) [inline sdk/Pulumi.Automation/WorkspaceStack.cs:502]: Ick I did not realize how grim this was to set on the command line. Very good reason to support this in automation api.

**Review by Frassle — APPROVED** [review]
Thanks, change looks good to me.

**Pull request merged**